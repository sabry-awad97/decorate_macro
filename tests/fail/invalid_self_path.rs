use decorate_macro::decorate;

struct Test {
    value: i32,
}

impl Test {
    #[decorate("invalid.path")] // Should fail - doesn't start with 'self'
    fn test(&self) -> i32 {
        self.value
    }
}

fn main() {}
