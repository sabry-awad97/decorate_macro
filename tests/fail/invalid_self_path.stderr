error: Path must start with 'self'
 --> tests/fail/invalid_self_path.rs:8:5
  |
8 |     #[decorate("invalid.path")] // Should fail - doesn't start with 'self'
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `decorate` (in Nightly builds, run with -Z macro-backtrace for more info)
